packages/labs/src/scoped-registry/scoped-registry-mixin.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {LitElement} from 'lit-element';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type LitElementConstructor = new (...args: any[]) => LitElement;
+
+export type ScopedElementsMap = {
+  [key: string]: typeof HTMLElement;
+};
+
+export function UseScopedRegistry<SuperClass extends LitElementConstructor>(
+  superclass: SuperClass
+): SuperClass {
+  return class ScopedRegistryMixin extends superclass {
+    registry?: CustomElementRegistry;
+
+    /**
+     * Obtains the scoped elements definitions map
+     */
+    static get scopedElements(): ScopedElementsMap {
```

Comment:
I'd make this an optional property, and if it's not defined don't create a registry.

Reply:
We could have lazy definitions later and we can't bind the registry later to the shadowRoot, don't you think?

Reply:
Not sure about this. If it's not going to declare the registry, it doesn't make sense to use the mixin, no?

Reply:
I could see a custom base class add this mixin where not every element that uses it may use scopes.

Reply:
so in  case I want to add lazy definitions then I must override 'createRenderRoot` to add a registry per instance... makes sense. I didn't think about that. Thanks

Reply:
Done